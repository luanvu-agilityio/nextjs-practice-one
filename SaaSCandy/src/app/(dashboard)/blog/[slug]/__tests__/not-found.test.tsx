import React from 'react';
import BlogPostNotFound from '@/app/(dashboard)/blog/[slug]/not-found';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('BlogPostNotFound - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<BlogPostNotFound />);
    expect(container).toMatchSnapshot();
  });

  it('should render not found message', () => {
    render(<BlogPostNotFound />);

    expect(screen.getByText('Blog Post Not Found')).toBeInTheDocument();
    expect(
      screen.getByText(/The blog post you're looking for doesn't exist/)
    ).toBeInTheDocument();
  });

  it('should render navigation buttons with correct links', () => {
    render(<BlogPostNotFound />);

    const browsePostsLink = screen.getByText('Browse All Posts').closest('a');
    const homeLink = screen.getByText('Back to Home').closest('a');

    expect(browsePostsLink).toHaveAttribute('href', '/blog');
    expect(homeLink).toHaveAttribute('href', '/');
  });

  it('should render icon svg', () => {
    const { container } = render(<BlogPostNotFound />);

    const svg = container.querySelector('svg');
    expect(svg).toBeInTheDocument();
    expect(svg).toHaveClass('text-gray-400');
  });
});
