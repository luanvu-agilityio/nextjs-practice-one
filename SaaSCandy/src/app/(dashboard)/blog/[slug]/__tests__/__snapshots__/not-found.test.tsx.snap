// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogPostNotFound - Snapshot Tests should match snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 bg-white min-h-[60vh] flex items-center justify-center"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="text-center max-w-md mx-auto"
      >
        <div
          class="mb-6"
        >
          <div
            class="w-16 h-16 bg-gray-100 rounded-full flex items-center justify-center mx-auto mb-4"
          >
            <svg
              class="w-8 h-8 text-gray-400"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <h2
            aria-level="2"
            class="text-5xl font-semibold text-gray-900 mb-4"
            role="heading"
          >
            Blog Post Not Found
          </h2>
          <p
            class="text-md text-gray-600 mb-6"
            role="paragraph"
          >
            The blog post you're looking for doesn't exist or may have been moved. Let's get you back to reading great content!
          </p>
        </div>
        <div
          class="space-y-3"
        >
          <a
            class="block"
            href="/blog"
          >
            <button
              class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-9 py-5 w-full"
            >
              Browse All Posts
            </button>
          </a>
          <a
            class="block"
            href="/"
          >
            <button
              class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-white text-orange-background border border-orange-background hover:bg-orange-background hover:text-white focus-visible:ring-primary px-9 py-5 w-full"
            >
              Back to Home
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
